### ISSUE TITLE
Failed to load emoji font

### ISSUE THREAD

**Issue by gabrc52** [google/flutter-desktop-embedding#90]
Error message:
`---- failed to open </usr/share/fonts/noto/NotoColorEmoji.ttf> <0> as a font`

As a result, emojis look like boxes

**Comment by stuartmorgan** [comment]
Unlike the Roboto issue, I don't think this one is going to be as easy to deal with. According to https://www.google.com/get/noto/help/emoji/ the font doesn't work on macOS or Windows, so just bundling it isn't going to work. Although it looks like on macOS it's not trying to use it; I'm trying to figure out where that mapping is controlled. Possibly in the Flutter engine.

On Linux, does adding Noto Color Emoji from https://www.google.com/get/noto/ to the asset list for the example app resolve the problem?

**Comment by gabrc52** [comment]
I added it and changed it the default to it, now I'm getting a segfault

**Comment by stuartmorgan** [comment]
Since as far as I know it doesn't contain glyphs for non-emoji, I wouldn't expect making it default to work. Does packaging it without changing the Dart code resolve the issue?

**Comment by stuartmorgan** [comment]
I looked at this a bit more. I'm not seeing the "failed to open" message on my Linux machine on a trunk build; I'm not sure if something changed (e.g., due to the change in how OS override is specified), or if that just isn't reproducible for me.

I do see that emojis don't work though. If I bundle NotoColorEmoji they still don't work in normal Texts, but do if the style for the text explicitly sets that font (at which point non-emojis don't render, unsurprisingly). That suggests that this has to do with the details of how emoji fallback is handled on Linux; see for instance the discussion on https://github.com/flutter/flutter/issues/17786 where what sounds like a similar bug was fixed for Fuchsia on the OS level.

Someone will need to dig into the font handling code in Linux and/or Skia to figure out how to address this.
